deploy-board/deploy_board/webapp/env_views.py

```diff
@@ -790,19 +790,22 @@ def post_create_env(request):
     clone_stage_name = data.get("clone_stage_name")
     description = data.get('description')
     
-    external_id = environs_helper.create_identifier_for_new_stage(request, clone_env_name, clone_stage_name)
-    
     if clone_env_name and clone_stage_name:
         try:
+            external_id = environs_helper.create_identifier_for_new_stage(request, env_name, stage_name)
             common.clone_from_stage_name(request, env_name, stage_name, clone_env_name,
                                         clone_stage_name, description, external_id)
         except TeletraanException as detail:
-            try:
-                environs_helper.delete_nimbus_identifier(request, external_id)
-            except:
-                message = 'Failed to delete Nimbus identifier {}, Error Message: {}'.format(external_id, detail)
-                log.error(message)
+            if external_id:
+                try:
+                    environs_helper.delete_nimbus_identifier(request, external_id)
+                except:
+                    message = 'Also failed to delete Nimbus identifier {}. Please verify that identifier no longer exists, Error Message: {}'.format(external_id, detail)
+                    log.error(message)
+            else:
+                message = 'Failed to create identifier for {}/{}: {}'.format(env_name, stage_name, detail)
                 messages.add_message(request, messages.ERROR, message)
+                raise detail
```

Comment:
Should we raise exception in both cases?

Reply:
Outside the else block

Reply:
This method is actually not even being used since the Create Environment button brings you to Ninbus instead of the modal that uses this method. I do imagine if for some reason it is used in the future we would want to raise the Exception to that the env is not created without a Nimbus Id.

Reply:
It is used when a cluster is cloned. i.e. config/capacity/ -> Clone Cluster.